docs/configuration.rst

```diff
@@ -1,45 +1,50 @@
 Configuration
 -------------
-Configuration parameters can be set in the following ways:
+Individual parameters can be set in the following ways:
 
-- In JSON file(s)
 - Directly when instantiating a new API object
+- Through environment variables
+- Through environment variables configured in *.env* files
 
-Configuration parameters
-========================
+Configuration on instantiation
+==============================
 
-You can configure the SDK with the following parameters:
+Configuration on a per-object basis takes precedence over any other configuration source:
 
-- `api_key`: The API key created in https://portal.us-east-1.mbedcloud.com/ (Required)
-- `host`: The fully qualified hostname (schema, host, port) of the api server (Optional)
+.. code-block:: python
 
-Configuration files
-==========================
-The file should be named `.mbed_cloud_config.json` and contain JSON describing configuration options
+  from mbed_cloud import AccountManagementAPI
+  config = { "api_key": "ak_*********" }
+  api = AccountManagementAPI(config)
 
-.. code-block:: text
+Production deployment
+=====================
+Many CI and production server providers allow direct configuration of environment variables
+in a secure manner, to the equivalent of:
 
-  $ cat .mbed_cloud_config.json
-  {
-    "api_key": "ak_*********"
-  }
+``export MBED_CLOUD_SDK_API_KEY=ak_abcdef123``
 
-In case of multiple configs, precedence for conflicting parameters is given to the more specific config.
-In order of increasing specificity, config file(s) should be placed in one of the following directories:
+Local development
+=================
+When developing locally, a `.env` file excluded from version control can be used to
+configure the SDK without modifying your system's environment:
 
-- `/etc/`
-- current user home
-- current working directory
-- the path specified by the environment variable: `MBED_CLOUD_SDK_CONFIG`
+*(in file: "/path/to/my/project/.env")*
 
+``MBED_CLOUD_SDK_API_KEY=ak_abcdef123``
 
-Configuration on instantiation
-==============================
+The file should be named `.env` and placed at any level on your directory tree between
+your project's working directory and the root.
+Files closer to the working directory, and any variables in the system environment, take priority.
 
-Configuration on a per-object basis takes precedence over any other configuration source:
+Configuration parameters
+========================
 
-.. code-block:: python
+You can configure the SDK with the following parameters:
 
-  from mbed_cloud import AccountManagementAPI
-  config = { "api_key": "ak_*********" }
-  api = AccountManagementAPI(config)
+=========== ========================== =================================================================== ========
+parameter   env var                    description                                                         optional
```

Comment:
Please capitalise title and the long form of `env var` would be easier to consume.